skimage/registration/_tvl1.py

```diff
@@ -0,0 +1,206 @@
+
+        rho_0 = wI1 - I0 - (grad*flow0).sum(0)
+
+        for _ in range(niter):
+
+            # Data term
+
+            rho = rho_0 + (grad*flow).sum(0)
+
+            idx = abs(rho) <= f0 * NI
+
+            flow_ = flow
+
+            flow_[:, idx] -= rho[idx]*grad[:, idx]/NI[idx]
+
+            idx = ~idx
+            srho = f0 * np.sign(rho[idx])
+            flow_[:, idx] -= srho*grad[:, idx]
+
+            # Regularization term
+            flow = flow_.copy()
+
+            for idx in range(flow.shape[0]):
+                s_p[0] = idx
+                for _ in range(2):
+                    for ax in range(flow.shape[0]):
+                        s_g[0] = ax
+                        s_g[ax+1] = slice(0, -1)
+                        g[tuple(s_g)] = np.diff(flow[idx], axis=ax)
+                        s_g[ax+1] = slice(None)
+
+                    norm = np.sqrt((g ** 2).sum(0))[np.newaxis, ...]
+                    norm *= f1
+                    norm += 1.
+                    proj[idx] -= dt * g
+                    proj[idx] /= norm
+
+                    # d will be the (negative) divergence of p[idx]
+                    d = -proj[idx].sum(0)
+                    for ax in range(flow.shape[0]):
+                        s_p[1] = ax
+                        s_p[ax+2] = slice(0, -1)
+                        s_d[ax] = slice(1, None)
+                        d[tuple(s_d)] += proj[tuple(s_p)]
+                        s_p[ax+2] = slice(None)
+                        s_d[ax] = slice(None)
+
+                    flow[idx] = flow_[idx] + d
+
+        flow0 -= flow
+        if (flow0*flow0).sum() < tol:
+            break
+
+        flow0 = flow
+
+    return flow
+
+
+def tvl1(I0, I1, dt=0.2, lambda_=15, tau=0.3, nwarp=5, niter=10,
+         tol=1e-4, prefilter=False):
+    """Coarse to fine TV-L1 optical flow estimator.
+
+    TV-L1 ia popular algorithm for optical flow estimation intrudced
+    by Zack et al. [1]_, improved in [2]_ and detailed in [3]_.
+
+    Parameters
+    ----------
+    I0 : ~numpy.ndarray
+        The first gray scale image of the sequence.
+    I1 : ~numpy.ndarray
+        The second gray scale image of the sequence.
+    dt : float
+        Time step of the numerical scheme. Convergence is proved for
+        values dt < 0.125, but it can be larger for faster
+        convergence.
+    lambda_ : float
+        Attachement parameter. The smaller this parameter is,
+        the smoother is the solutions.
+    tau : float
+        Tightness parameter. It should have a small value in order to
+        maintain attachement and regularization parts in
+        correspondence.
+    nwarp : int
+        Number of times I1 is warped.
+    niter : int
+        Number of fixed point iteration.
+    tol : float
+        Tolerance used as stopping criterion based on the L² distance
+        between two consecutive values of (u, v).
+    prefilter : bool
+        whether to prefilter the estimated optical flow before each
```

Comment:
```suggestion
        Whether to prefilter the estimated optical flow before each
```

A few more words about this parameter would be nice (when is it useful?)

Reply:
This option is proposed in the improved version of the algorithm [2]. It is not really necessary if the attachment parameter is low enough to produce a smooth result.